hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -435,20 +436,21 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
     // validate that we can support partial updates
     if (writePartialUpdates) {
       if (hoodieCatalogTable.tableConfig.getBootstrapBasePath.isPresent) {
-        throw new UnsupportedOperationException("Partial updates are not supported for bootstrap tables.")
+        throw new HoodieNotSupportedException("Partial updates are not supported for bootstrap tables.")
       }
 
       if (!hoodieCatalogTable.tableConfig.populateMetaFields()) {
-        throw new UnsupportedOperationException("Partial updates are not supported for virtual key tables.")
+        throw new HoodieNotSupportedException("Partial updates are not supported for virtual key tables.")
       }
 
-      if (HoodieAvroUtils.containsUnsupportedTypesForFileGroupReader(fullSchema)) {
-        throw new UnsupportedOperationException("Partial updates are not supported for tables with enum columns")
+      val tableSchemaResolver = new TableSchemaResolver(hoodieCatalogTable.metaClient)
```

Comment:
Could we use `hoodieCatalogTable.tableSchema`?

Reply:
Let's see if we can avoid re-reading the table schema.

Reply:
No. That is a spark schema, not an avro schema